src/deadline_worker_agent/sessions/session.py

```diff
@@ -1208,16 +1256,17 @@ def _handle_action_update(
                 fail_message="TIMEOUT - Exceeded the allotted runtime limit.",
             )
 
-        self._report_action_update(
-            SessionActionStatus(
-                id=current_action.definition.id,
-                status=action_status,
-                start_time=current_action.start_time,
-                end_time=now if action_status.state != ActionState.RUNNING else None,
-                update_time=now if action_status.state == ActionState.RUNNING else None,
-                completed_status=completed_status,
+        if not self._output_sync_target_action:
+            self._report_action_update(
+                SessionActionStatus(
+                    id=current_action.definition.id,
+                    status=action_status,
+                    start_time=current_action.start_time,
+                    end_time=now if action_status.state != ActionState.RUNNING else None,
+                    update_time=now if action_status.state == ActionState.RUNNING else None,
+                    completed_status=completed_status,
+                )
             )
```

Comment:
I understand this is because we don't want to complete the action

This will mean that progress updates for output upload will not be reported back in `UpdateWorkerSchedule` API requests. I don't believe progress reporting is supported by the output upload yet, but maybe this should be added as a comment for when we build that in.